tools/autograd/gen_trace_type.py

```diff
@@ -176,8 +175,12 @@ def dispatch_trace_input(
     if f.func.is_out_fn():
         # for *_out functions, handle the result argument differently for inplace/outplace.
         # For inplace: just add the input to the end to confirm with the JIT schema
-        name = f.func.arguments.out[0].name  # TODO: old codegen behavior - should fix
-        inplace = ADD_TRACE_INPUT.substitute(name=name, input=name)
+        inplace = [
+            ADD_TRACE_INPUT.substitute(
+                name=f.func.arguments.out[i].name, input=f.func.arguments.out[i].name
+            )
+            for i in range(num_out_args)
+        ]
```

Comment:
Since `inplace` is a list now, you may consider to revise the code below to consider that:
original code
```python
        trace_inputs = itertools.chain(
            trace_inputs,
            [
                SELECT.substitute(
                    cond="tracer_state->force_outplace",
                    true="\n".join(outplace),
                    false=inplace,
                )
            ],
        )
```
```python
        trace_inputs = itertools.chain(
            trace_inputs,
            [
                SELECT.substitute(
                    cond="tracer_state->force_outplace",
                    true="\n".join(outplace),
                    false="\n".join(inplace),
                )
            ],
        )
```

Reply:
yeah, I have fixed it. thank you.